presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1758,24 +1759,42 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
                 .map(HiveColumnHandle.class::cast)
                 .collect(toImmutableMap(HiveColumnHandle::getName, Functions.identity()));
 
+        SchemaTableName tableName = ((HiveTableHandle) tableHandle).getSchemaTableName();
         return new ConnectorPushdownFilterResult(
                 getTableLayout(
                         session,
                         new HiveTableLayoutHandle(
-                                ((HiveTableHandle) tableHandle).getSchemaTableName(),
-                                ImmutableList.copyOf(hivePartitionResult.getPartitionColumns()),
+                                tableName,
+                                hivePartitionResult.getPartitionColumns(),
```

Comment:
This is the responsibility of the constructor to do the defensive copy. 

Actually in this place it was used for different reason. `ImmutableList.copyOf(...)` has this not very well know, but very useful semantic that can be used for up casting generic collections:

```
 <E> ImmutableList<E> copyOf(Collection<? extends E> elements)
```

If `ChildClass` extends `ParentClass`, casting `List<ChildClass>` to `List<ParentClass>` is not correct. As after casting to  `List<ParentClass>` you are allowed to add elements that are not of the `List<ChildClass>`. But, since you cannot modify `ImmutableList`, it is fine to cast `ImmutableList<ChildClass>` to `ImmutableList<ParentClass>`. And this semantic was used in this place mainly for this reason.

After i changed the `List<ColumnHandle> partitionColumns` to `List<HiveColumnHandle> partitionColumns` this change is no longer needed. But as it is not needed - I'm simply going to revert it.